src/common/com/intellij/plugins/haxe/lang/util/HaxeAstUtil.java

```diff
@@ -128,7 +128,13 @@ public static boolean isInteger(@Nullable String s) {
   public static boolean isNumber(@Nullable ASTNode t) {
     if (t != null) {
       String text = t.getText();
-      return isInteger(text) || isFloat(text);
+      // Optimization: If it doesn't start with '+', '-', or digit, it's not a number (ignoring 'Nan' and Infinity).
+      // Using the Float class to detect numbers is fairly expensive.  Before this optimization, about 2/3 of the
+      // time we spent evaluating a conditional expression was spent in isFloat().
+      char firstChar = text.charAt(0);
+      if (Character.isDigit(firstChar) || '+' == firstChar || '-' == firstChar) {
```

Comment:
Float could be written as `.025` that equals `0.025`

Reply:
Ah, yes.  Thanks!